swarm/storage/localstore.go

```diff
@@ -83,8 +94,19 @@ func (self *LocalStore) CacheCounter() uint64 {
 }
 
 // LocalStore is itself a chunk store
-// unsafe, in that the data is not integrity checked
 func (self *LocalStore) Put(chunk *Chunk) {
+	valid := true
+	for _, v := range self.Validators {
+		if v.Validate(chunk.Key, chunk.SData) {
```

Comment:
actually @zelig is right - this is indeed wrong, valid is never set to `false`. I don't think this is what we discussed earlier.

1. By default `valid` should be `true`, so that if there are no validators the chunk is marked as valid.
2. All validators should mark `valid` with `true` if they succeed and break, because if a single validator passes, it means that the chunk is valid. (this condition is present in the current version)
3. All validator should mark `valid` with `false` if they fail, but NOT break, because there might be other validators down the line that might succeed.

Looks like we are missing 3. at the moment.